ui-admin/src/study/surveys/FormOptions.tsx

```diff
@@ -0,0 +1,163 @@
+import Api, { Survey, VersionedForm } from 'api/api'
+import Modal from 'react-bootstrap/Modal'
+import { Button } from 'components/forms/Button'
+import { instantToDefaultString } from 'util/timeUtils'
+import Select from 'react-select'
+import { faArrowRightFromBracket, faDownload } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { StudyEnvContextT, studyEnvFormsPath } from '../StudyEnvironmentRouter'
+import { useLoadingEffect } from 'api/api-utils'
+import { saveBlobAsDownload } from 'util/downloadUtils'
+import { SaveableFormProps } from './SurveyView'
+import { DocsKey, ZendeskLink } from '../../util/zendeskUtils'
+import InfoPopup from '../../components/forms/InfoPopup'
+
+/** component for selecting versions of a form */
+export default function FormOptions({
+  studyEnvContext, workingForm, updateWorkingForm, onDismiss, isDirty,
+  visibleVersionPreviews, setVisibleVersionPreviews, isConsentForm = false
+}:
+                                          {studyEnvContext: StudyEnvContextT, workingForm: VersionedForm,
+                                            isDirty: boolean,
+                                            updateWorkingForm: (props: SaveableFormProps) => void,
+                                            onDismiss: () => void
+                                            visibleVersionPreviews: VersionedForm[],
+                                            setVisibleVersionPreviews: (versions: VersionedForm[]) => void,
+                                          isConsentForm?: boolean}) {
+  const stableId = workingForm.stableId
+
+  const downloadJSON = () => {
+    const content = workingForm.content
+    // To get this formatted nicely and not as one giant line, need to parse
+    // this as an object and then stringify the result.
+    const blob = new Blob(
+      [JSON.stringify(JSON.parse(content), null, 2)],
+      { type: 'application/json' })
+    const filename = `${stableId}_v${workingForm.version}${isDirty ? '_draft' : ''}.json`
+    saveBlobAsDownload(blob, filename)
+  }
+  const isSurvey = !!(workingForm as Survey).surveyType
+
+  return <Modal show={true} onHide={onDismiss}>
+    <Modal.Header closeButton>
+      <Modal.Title>{workingForm.name} <InfoPopup placement="bottom" content={<div>
+        See the &quot;Options&quot; section in
+        our <ZendeskLink doc={DocsKey.SURVEY_EDIT}>survey editing docs</ZendeskLink> for
+        more detail.  Note that you need to  &quot;Save&quot; the survey
+        for any changes to options on this page to take effect.
+      </div>}/></Modal.Title>
+    </Modal.Header>
+    <Modal.Body>
+      { isSurvey &&
+        <label className="form-label d-block">
+          <input type="checkbox" checked={(workingForm as Survey).required}
+            onChange={e => updateWorkingForm({
+              ...workingForm, required: e.target.checked
+            })}
+          /> Required
```

Comment:
I made a required survey but it didn't update when I went back to the options; it was just unchecked.

Reply:
hmm... we'll have to do a screenshare.  It works great on my machine :)

Reply:
@connorlbark did you also save the survey afterwards? That part tripped me up- I forgot to save after toggling.

Requiring a save is consistent with other parts of self-service, so I don't think the behavior should necessarily change right now, but I think I'd advocate for the information in the tooltip to be on the modal itself to make it more clear.